components/collapsible-panel/collapsible-panel.js

```diff
@@ -0,0 +1,405 @@
+import '../colors/colors.js';
+import '../icons/icon.js';
+import '../icons/icon-custom.js';
+import '../expand-collapse/expand-collapse-content.js';
+import { css, html, LitElement } from 'lit';
+import { heading1Styles, heading2Styles, heading3Styles, heading4Styles } from '../typography/styles.js';
+import { classMap } from 'lit/directives/class-map.js';
+import { offscreenStyles } from '../offscreen/offscreen.js';
+import { RtlMixin } from '../../mixins/rtl-mixin.js';
+
+const defaultHeading = 3;
+
+/**
+ * A container with a title that can be expanded/collapsed to show/hide content.
+ * @slot header - Slot for supporting header content
+ * @slot summary - Slot for the summary of the expanded content. Only accepts `d2l-collapsible-panel-summary-item`
+ * @slot default - Slot for the expanded content
+ * @slot actions - Slot for buttons and dropdown openers to be placed in top right corner of header
+ * @fires d2l-collapsible-panel-expand - Dispatched when the panel is expanded
+ * @fires d2l-collapsible-panel-collapse - Dispatched when the panel is collapsed
+ */
+class CollapsiblePanel extends RtlMixin(LitElement) {
+
+	static get properties() {
+		return {
+			/**
+			 * REQUIRED: The title of the panel
+			 * @type {string}
+			 */
+			title: { type: String },
+			/**
+			 * The semantic heading level (h1-h4)
+			 * @type {'1'|'2'|'3'|'4'}
+			 * @default "3"
+			 */
+			headingLevel: { attribute: 'heading-level', type: String, reflect: true },
+			/**
+			 * The heading style to use
+			 * @type {'1'|'2'|'3'|'4'}
+			 * @default "3"
+			 */
+			headingStyle: { attribute: 'heading-style', type: String, reflect: true },
+			/**
+			 * Whether or not the panel is expanded
+			 * @type {boolean}
+			 */
+			expanded: { type: Boolean, reflect: true },
+			/**
+			 * REQUIRED: Label describing the contents of the header.
+			 * Used for screen readers.
+			 * @type {string}
+			 */
+			expandCollapseLabel: { attribute: 'expand-collapse-label', type: String, reflect: true },
+			/**
+			 * Type of collapsible panel
+			 * @type {'default'|'subtle'|'inline'}
+			 * @default "default"
+			 */
+			type: { type: String, reflect: true },
+			/**
+			 * Horizontal padding of the panel
+			 * @type {'default'|'large'}
+			 * @default "default"
+			 */
+			padding: { type: String, reflect: true },
+		};
+	}
+
+	static get styles() {
+		return [heading1Styles, heading2Styles, heading3Styles, heading4Styles, offscreenStyles, css`
+			:host {
+				--d2l-collapsible-panel-focus-outline: solid 2px var(--d2l-color-celestine);
+				--d2l-collapsible-panel-spacing-inline: 0.9rem;
+				--d2l-collapsible-panel-header-spacing: 0.6rem;
+				--d2l-collapsible-panel-transition-time: 0.2s;
+				--d2l-collapsible-panel-arrow-time: 0.5s;
+				display: block;
+			}
+			.d2l-collapsible-panel {
+				border: 1px solid var(--d2l-color-mica);
+				border-radius: 0.3rem;
+				display: block;
+			}
+			:host([padding=large][type=inline]) .d2l-collapsible-panel {
```

Comment:
Nit: since these are strings I'd personally prefer `:host([padding="large"][type="inline"])`. And due to the way variables cascade, I think you can actually drop the `.d2l-collapsible-panel` part and just apply the variable value to the host itself.

Reply:
ahh, good catch on the host! missed this when I refactored the inline spacing to a css variable